sklearn/cluster/_spectral.py

```diff
@@ -258,9 +258,14 @@ def spectral_clustering(
       Ulrike von Luxburg
       http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.165.9323
 
-    - Multiclass spectral clustering, 2003
-      Stella X. Yu, Jianbo Shi
-      https://www1.icsi.berkeley.edu/~stellayu/publication/doc/2003kwayICCV.pdf
+    - Modern Preconditioned Eigensolvers for Spectral Image Segmentation
```

Comment:
After looking at the papers themselves, I am not inclined to include them. They have little added value for the reader.
This said, a reference to the LOBPCG algorithm [1] with a cross-reference in the `eigen_solver` would be meaningful. 

[1] A. V. Knyazev (2001), Toward the Optimal Preconditioned Eigensolver: Locally Optimal Block Preconditioned Conjugate Gradient Method. SIAM Journal on Scientific Computing 23, no. 2, pp. 517-541. DOI:10.1137/S1064827500366124

Reply:
@glemaitre These are the first publications specifically describing and testing the use of LOBPCG (2003) and LOBPCG/AMG (2006) for spectral clustering and image segmentation. The 2001 paper does not even mention spectral clustering or image segmentation to my recollection, so I am unsure if appropriate in this file.

Reply:
If we refer to Spectral Clustering used for image segmentation then we should refer to the oldest reference, independently from the solver. However, I think that this worth to have a link to the solver used  for the `solver` parameter and option, in the same spirit of my other comment: https://github.com/scikit-learn/scikit-learn/pull/20987#discussion_r709029962

Reply:
OK, will make the change as you suggest